Modules/xSharePoint/DSCResources/MSFT_xSPFeature/MSFT_xSPFeature.psm1

```diff
@@ -47,23 +49,55 @@ function Set-TargetResource
         [parameter(Mandatory = $true)]  [System.String] $Url,
         [parameter(Mandatory = $true)]  [ValidateSet("Farm","WebApplication","Site","Web")] [System.String] $FeatureScope,
         [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount,
-        [parameter(Mandatory = $true)]  [ValidateSet("Present","Absent")] [System.String] $Ensure
+        [parameter(Mandatory = $true)]  [ValidateSet("Present","Absent")] [System.String] $Ensure,
+        [parameter(Mandatory = $false)] [System.String] $Version
     )
 
     $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
         $params = $args[0]
         
 
-        $runParams = @{}
-        $runParams.Add("Identity", $params.Name)
+        $runParams = @{ 
+            Identity = $params.Name 
+            Verbose = $false
+        }
+
         if ($params.FeatureScope -ne "Farm") {
             $runParams.Add("Url", $params.Url)
         }
 
+        # Parameters for get
+        $checkParams = @{ 
+            Identity = $params.Name 
+            Verbose = $false    
+        }
+        
+        if ($params.FeatureScope -eq "Farm") 
+        {
+            $checkParams.Add($params.FeatureScope, $true)
+        } 
+        else
+        {
+            $checkParams.Add($params.FeatureScope, $params.Url)
+        }
+        
+
         if ($params.Ensure -eq "Present") {
+            if (Get-SPFeature @checkParams -ErrorAction SilentlyContinue){
+                    
+                # Disable the feature first if it already exists.
+                $runParams.Add("Confirm", $false)    
+                Write-Verbose "Disable Feature '$($params.Name)' because it is already active at scope '$($params.FeatureScope)'..."
+                Disable-SPFeature @runParams
+            }
```

Comment:
I'm assuming this is the logic to disable the feature and reactivate where the version is different, yea? If so, rather than just assuming that it's there can we ensure that the version number is different here too (which if you use the Get-TargetResource method above you'll have easy access to the current values)